### PR TITLE
fix(installer): trim EKS cluster name from both ARN and eksctl formats

### LINKED ISSUE #39482: cilium-cli fails to detect EKS cluster name with eksctl.io context format
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.17.3 and lower than v1.18.0

### Bug description

cilium install fails to detect a proper cluster name from kubeconfig when using certain EKS contexts.

Specifically, when the current kubeconfig context has the format:
`<cluster-name>.<region>.eksctl.io`

The auto-detected cluster name becomes:
`<cluster-name>.<region>.eksctl.io`
which often exceeds the maximum length (32 chars) and causes cilium install to fail.

Example :
```
$ kubectl config current-context
zzuckerfrei@eks-skuber-flash-3.ap-northeast-1.eksctl.io

$ cilium install --version 1.17.3 \
  --helm-set hubble.relay.enabled=true \
  --helm-set hubble.ui.enabled=true \
  --set hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}" \
  --set cluster.id=1
🔮 Auto-detected Kubernetes kind: EKS
ℹ️  Using Cilium version 1.17.3
🔮 Auto-detected cluster name: eks-skuber-flash-3-ap-south-1-eksctl-io
🔮 Auto-detected kube-proxy has been installed
🔥 Patching the "aws-node" DaemonSet to evict its pods...

Error: Unable to install Cilium: execution error at (cilium/templates/validate.yaml:143:5): The cluster name is invalid: must not be more than 32 characters
```
The cluster name exceeds the maximum length, resulting in installation failure.

### How can we reproduce the issue?

1. Create EKS cluster using eksctl with a cluster name over ~16 characters 
(so the context name exceeds 32 characters after .region.eksctl.io suffix is added):
`eksctl create cluster --name <cluster-name> --region <region>`
Example :
`eksctl create cluster --name eks-skuber-flash-3 --region ap-northeast-1`

2. Check kubeconfig context:
`kubectl config current-context`

3. Run:
`cilium install --version 1.17.3 --set cluster.id=1`

4. Observe the auto-detected cluster name.

### Cilium Version

```
$ cilium version
cilium-cli: v0.18.3 compiled with go1.24.2 on darwin/arm64
cilium image (default): v1.17.2
cilium image (stable): v1.17.3
```

### Kernel Version

Kernel version : 5.10.236-227.928.amzn2.x86_64
Architecture : amd64
OS : Amazon Linux 2

### Kubernetes Version

v1.32

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

```shell

```

### Anything else?

I initially suspected this was due to `eksctl` version differences, but after testing across multiple versions (`0.188.0` to `0.207.0`), I confirmed the kubeconfig context format is **not strictly tied to the eksctl version**. 
It seems to depend on other environmental factors.
Even within the same AWS account, some users get ARN-form context names, others get .eksctl.io names.

### Proposed fix
I propose improving the trimEKSClusterARN logic to handle both:

- ARN-based cluster names (`arn:aws:eks:<region>:<account>:cluster/<cluster-name>`)
- eksctl context-based names (`<cluster-name>.<region>.eksctl.io`)

I would be happy to submit a PR implementing this change if the maintainers are open to it. 
Please let me know if you’d like me to proceed!

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zzuckerfrei**
Clusters created with `eksctl` often use a kube-context like `<cluster-name>.<region>.eksctl.io`, 
which was previously not correctly trimmed during auto-detection.

This led to validation errors such as:
`cluster name must not be more than 32 characters`.

This commit introduces `trimEKSClusterName()` to correctly handle both:

- ARN format: `arn:aws:eks:<region>:<account>:cluster/<name>`

- eksctl FQDN format: `<name>.<region>.eksctl.io`

Changes include:

- Applying trimming logic in `autodetectAndValidate()` before replacing disallowed characters (`_`, `.`, `:`)

- Adding unit tests for edge cases across both formats

This ensures consistent and robust cluster name parsing across different EKS setups.

Fixes: #39482

```release-note
Improve EKS cluster name auto-detection by supporting both ARN and eksctl FQDN formats to avoid validation errors caused by overly long names.
```

### CODE REVIEW

File: cilium-cli/install/autodetect.go [hunk 1]
```diff
@@ -74,18 +74,27 @@ func getClusterName(helmValues map[string]any) string {
 	return clusterName
 }
 
-func trimEKSClusterARN(fullARN string) string {
+func trimEKSClusterName(identifier string) string {
+	// Handle ARN format: arn:aws:eks:<region>:<account>:cluster/<cluster-name>
 	const prefix = "cluster/"
-	idx := strings.LastIndex(fullARN, prefix)
```
Comment (squeed) [inline cilium-cli/install/autodetect.go:78]: minor idea: should `prefix` be `:cluster/` just to make it a bit more reliable? (I know you didn't initially touch this)

File: cilium-cli/install/autodetect.go [hunk 2]
```diff
@@ -74,18 +74,27 @@ func getClusterName(helmValues map[string]any) string {
 	return clusterName
 }
 
-func trimEKSClusterARN(fullARN string) string {
+func trimEKSClusterName(identifier string) string {
```
Comment (squeed) [inline cilium-cli/install/autodetect.go:81]: Now that you understand this function more than anyone else in the world, how about a quick docblock :-)

**Comment by squeed** [comment]
Looks reasonable! Just a few small requests, nothing exciting. Thanks for the PR!

(feel free to re-request review, otherwise it gets lost in the noise.

**Comment by zzuckerfrei** [comment]
> Looks reasonable! Just a few small requests, nothing exciting. Thanks for the PR!
> 
> (feel free to re-request review, otherwise it gets lost in the noise.)

Thanks for the review! 🙏

I'll update the prefix to `:cluster/` and add a docblock for `trimEKSClusterName()` as suggested.

**Review by squeed — APPROVED** [review]

**Comment by julianwiedmann** [comment]
Looking at the CI results, this will need a rebase (to pick up some multi-pool IPAM changes that are tested by the workflow). Will do so now ...

**Comment by squeed** [comment]
The unit test you added is [failing](https://github.com/cilium/cilium/actions/runs/15164948467/job/42643142691):

```
--- FAIL: Test_trimEKSClusterName (0.00s)

    autodetect_test.go:43: 
        	Error Trace:	/home/runner/work/cilium/cilium/cilium-cli/install/autodetect_test.go:43
        	Error:      	Not equal: 
        	            	expected: "my-cluster"
        	            	actual  : "cluster/my-cluster"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-my-cluster
        	            	+cluster/my-cluster
        	Test:       	Test_trimEKSClusterName
```

**Comment by zzuckerfrei** [comment]
Hi, I noticed a test case failed after changing the prefix from `"cluster/"` to `":cluster/"`.

The failed test case uses `"cluster/my-cluster"` as the cluster name, which doesn’t seem to reflect a common EKS naming pattern.

Given this, I’m considering a few possible directions:

1. **Update the test case** to use a more realistic EKS-style input, such as an ARN (e.g., `arn:aws:eks:...:cluster/my-cluster`) or an eksctl FQDN (e.g., `my-cluster.region.eksctl.io`).

2. **Reconsider the prefix change**, and possibly go back to using `"cluster/"` if that better supports legacy or edge cases.

3. **Remove the failing test case** if we think the input format is too unlikely to support.

I’d really appreciate your thoughts on which direction would be best here. 🙂


> The unit test you added is [failing](https://github.com/cilium/cilium/actions/runs/15164948467/job/42643142691):
> 
> ```
> --- FAIL: Test_trimEKSClusterName (0.00s)
> 
>     autodetect_test.go:43: 
>         	Error Trace:	/home/runner/work/cilium/cilium/cilium-cli/install/autodetect_test.go:43
>         	Error:      	Not equal: 
>         	            	expected: "my-cluster"
>         	            	actual  : "cluster/my-cluster"
>         	            	
>         	            	Diff:
>         	            	--- Expected
>         	            	+++ Actual
>         	            	@@ -1 +1 @@
>         	            	-my-cluster
>         	            	+cluster/my-cluster
>         	Test:       	Test_trimEKSClusterName
> ```

**Comment by squeed** [comment]
@zzuckerfrei let's update the test case to reflect a real cluster naming convention :-). But great question.

**Comment by sayboras** [comment]
Seems like the unrelated groups/reviewers are added due to some reasons (e.g. wrong target branch, out of date local, etc), I just removed all, and re-add cilium-cli group

**Comment by zzuckerfrei** [comment]
> Seems like the unrelated groups/reviewers are added due to some reasons (e.g. wrong target branch, out of date local, etc), I just removed all, and re-add cilium-cli group

Hi, sorry about the noise earlier — I mistakenly pushed a broken rebase that included a huge number of unrelated changes and reviewers.

Thanks a lot for cleaning up the reviewers list and for your patience!

**Review by tklauser — CHANGES_REQUESTED** [review]
@zzuckerfrei could you please squash the three commits into a single one? Other than that LGTM.

**Review by squeed — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
Thanks!

**Comment by tklauser** [comment]
Rebased to pull in fixes for the CI failures.

**Comment by zzuckerfrei** [comment]
Hi, @tklauser,

Thank you so much for the rebase and all your help with this PR! 🙏 

I noticed there are still some CI failures. Is there anything I can do on my end to help resolve them, or should I wait for the infrastructure issues to be resolved?

Really appreciate your time and patience!

**Comment by tklauser** [comment]
> I noticed there are still some CI failures. Is there anything I can do on my end to help resolve them, or should I wait for the infrastructure issues to be resolved?

The failures look unrelated to this change and are likely flaky tests. I've re-triggered the failing workflows.
